tests/integration/suite/daprd/workflow/crossapp/appdown.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2025 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crossapp
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/framework/process/http/app"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/process/scheduler"
+	"github.com/dapr/dapr/tests/integration/framework/process/sqlite"
+	"github.com/dapr/dapr/tests/integration/suite"
+	"github.com/dapr/durabletask-go/api"
+	"github.com/dapr/durabletask-go/backend"
+	"github.com/dapr/durabletask-go/client"
+	"github.com/dapr/durabletask-go/task"
+)
+
+func init() {
+	suite.Register(new(appdown))
+}
+
+// appdown tests error handling when a target app goes down during execution
+type appdown struct {
+	daprd1 *daprd.Daprd
+	daprd2 *daprd.Daprd
+	place  *placement.Placement
+	sched  *scheduler.Scheduler
+
+	registry1 *task.TaskRegistry
+	registry2 *task.TaskRegistry
+}
+
+func (a *appdown) Setup(t *testing.T) []framework.Option {
+	a.place = placement.New(t)
+	a.sched = scheduler.New(t,
+		scheduler.WithLogLevel("debug"))
+	db := sqlite.New(t,
+		sqlite.WithActorStateStore(true),
+		sqlite.WithMetadata("busyTimeout", "10s"),
+		sqlite.WithMetadata("disableWAL", "true"),
+	)
+
+	app1 := app.New(t)
+	app2 := app.New(t)
+
+	// Create registries for each app
+	a.registry1 = task.NewTaskRegistry()
+	a.registry2 = task.NewTaskRegistry()
+
+	appID1 := uuid.New().String()
+	appID2 := uuid.New().String()
+
+	// App2: Activity that will be called before the app goes down
+	a.registry2.AddActivityN("ProcessData", func(ctx task.ActivityContext) (any, error) {
+		var input string
+		if err := ctx.GetInput(&input); err != nil {
+			return nil, fmt.Errorf("failed to get input in app2: %w", err)
+		}
+
+		// Sleep for a while to ensure the workflow is actively executing when app2 is stopped
+		time.Sleep(10 * time.Second)
```

Comment:
Please do not use sleep. Use channels to signal processes are at different checkpoints.